app/helpers/form_helper.rb

```diff
@@ -3,4 +3,17 @@ def list_of_organisations
     @list_of_organisations ||=
       [OpenStruct.new(name: "", id: ""), Organisation.sorted_by_name].flatten
   end
+
+  def list_of_user_roles
+    @list_of_user_roles ||= build_user_roles_list
+  end
+
+  private
+
+  def build_user_roles_list
+    user_roles = User.roles.values
```

Comment:
You can write this as:

```ruby
User.roles.map{ |id, name| OpenStruct.new(id: id, name: name) }
```

`map` loops over each `User.roles` in turn. As it is a `Hash` the block accepts a `key` and `value`, that I've renamed to `id` and `name` for clarity.